### PR TITLE
Osong/resource sync

### PR DESCRIPTION
**Pull request by 1298se**
Added resource sync for #96. Reimplemented concurrently for serving firebase functions locally.

### CODE REVIEW

File: contrail-ui/src/store/actions/resourceActions.ts [hunk 1]
```diff
@@ -11,21 +13,49 @@ export const fetchRootResources =
...
+export const setResourceListener =
+    (): ThunkAction<Promise<any>, {}, null, actions.IResourceFetchAllAction> =>
+        (dispatch) => {
+            return new Promise(async (resolve, reject) => {
+                const user = authRef.currentUser;
+                if (user == null) {
+                    reject("current user is null");
+                    return;
+                }
+                const doc = dbRef.collection("users").doc(`${user.uid}`).collection("root").doc("resources");
```
Comment (virajg224) [inline contrail-ui/src/store/actions/resourceActions.ts]: why is it a string template

File: contrail-ui/src/components/resources/ResourceFrame.tsx [hunk 2]
```diff
@@ -12,13 +12,29 @@ import ResourceToolBar from "./resource-tool-bar/ResourceToolBar";
...
                 this.props.setSnackbarDisplay("error", "Failed to load resources: " + error.response);
             });
+        this.props.setResourceListener()
+        .then((unsubscribe) => {
+            this.setState({
+                unsubscribeListener: unsubscribe,
+            });
+        })
+        .catch((error) => {
+            this.props.setSnackbarDisplay("error", "Failed to fetch resources: " + error.response);
```
Comment (virajg224) [inline contrail-ui/src/components/resources/ResourceFrame.tsx]: change fetch to sync

File: contrail-ui/src/store/actions/resourceActions.ts [hunk 3]
```diff
@@ -11,21 +13,49 @@ export const fetchRootResources =
...
+                    .then((response) => {
+                        if (response.status === 200) {
+                            dispatch({
+                                type: constants.RESOURCE_FETCH_ALL,
+                                payload: response.data,
+                            });
+                            dispatch(setAppResourceLoadingState(false));
+                            resolve();
+                        } else {
+                            // TODO: Handle error response
```
Comment (virajg224) [inline contrail-ui/src/store/actions/resourceActions.ts]: add reject

File: contrail-ui/src/store/actions/resourceActions.ts (hunk 1 above)
Comment (1298se) [reply contrail-ui/src/store/actions/resourceActions.ts]: done

File: contrail-ui/src/components/resources/ResourceFrame.tsx (hunk 2 above)
Comment (1298se) [reply contrail-ui/src/components/resources/ResourceFrame.tsx]: done

**Comment by 1298se** [comment]
resolve #96

**Pull request merged**